02_bubble/alanhillgemann/main.go

```diff
@@ -0,0 +1,25 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprint(out, bubble([]int{3, 2, 1, 5}))
+}
+
+func bubble(arr []int) []int {
+	for i := 0; i <= len(arr)-1; i++ {
+		end := len(arr) - 2
+		for j := 0; j <= end; j++ {
```

Comment:
After the first iteration of the outer loop the largest element of the slice has bubbled up to the last index in the slice and therefore does not need to be included in any subsequent comparisons (it is already sorted).
Therefore the inner loop in bubble sort ranges from 0 to i:  `for j := 0; j < i; j++ {`

Reply:
Fixed